src/hooks/chart.ts

```diff
@@ -1090,4 +1127,113 @@ const updateWithTraceColData = (
       }
     }
   }
+
 };
+
+/**
+ * 
+ * @param result data object to be updated
+ * @param trace from plot configs
+ * @param item from response object
+ * @param textArray array for hover text
+ * @param index provided for heatmap case for corresponding y axis values
+ * @param templateParams provided for heatmap to access url_parameters
+ * @returns 
+ */
+const updateHoverTemplateData = (
+  result: any,  
+  trace: Trace, 
+  item: any, 
+  textArray: any[], 
+  alertFunctions?: any,
+  index: number=0, 
+  templateParams?: PlotTemplateParams,
+  ) => {
+  const hovertemplate_display_pattern = trace.hovertemplate_display_pattern; // use trace info
+  if(hovertemplate_display_pattern){
+    const validLink=ConfigService.ERMrest._renderHandlebarsTemplate(hovertemplate_display_pattern, { $self: { data: item }, 
+      $row: item, 
+      $url_parameters: templateParams?.$url_parameters
+    });
+    /**If there are any invalid params in the hover template display pattern, the link generated will be null.
+    Therefore the following piece of code will add an alert for stating that just once*/
+    if(!validLink && (Array.isArray(textArray[0]) ? textArray[0]?.length===1 : textArray?.length===1)){
+      if(alertFunctions.alerts?.length===0){
+          alertFunctions.addAlert('Invalid key provided for hover template display pattern!', ChaiseAlertType.WARNING);
+      }
+    }
+    const link = ConfigService.ERMrest._renderHandlebarsTemplate(hovertemplate_display_pattern, { $self: { data: item }, 
+      $row: item, 
+      $url_parameters: templateParams?.$url_parameters
+    }, null, {avoidValidation: true} );
+    if(link){
+      switch(result.type){
+        case 'bar':
+        case 'scatter':
+          textArray.push(link);
+          break;
+        case 'violin':
+          textArray.push(link);
+          break;
+        case 'pie':
+          textArray.push(link);
+          break;
+        case 'histogram':
+          textArray.push(extractAndFormatDate(link));
+          break;
+        case 'heatmap':
+          textArray[index].push(link);
+          break;
+        default:
+          break;   
+      }
+    }
+  }
+  return textArray;
+}
+
+/**
+ * Adds default hover text to the result object when hovertemplate_display_pattern is not configured 
+ * Note: This fixes the issue of displaying link in hover text when tick_display is configured for x or y axis 
+ * which was making the text hard to read in some plot types
+ * @param result trace object to be updated
+ */
+const defaultHoverTemplateDisplay = (
+  result: any,
+): void => {
+  const tempText: any[]=[];
+  if (result.data[0].type==='heatmap') {
+    result.data[0].hoverinfo= 'text';
+    result.data[0].z.forEach((zArr: any[],index: number)=>{
+      tempText.push([]);
```

Comment:
To avoid having multiple types (or using `any[]`), this whole block of code can be changed to:
```
const textArray = [];
zArr.forEach((val: any,i: number)=>{
  textArray.push(...)
})
tempText.push(textArray);
result.data[0].text=tempText;
```